app/src/main/java/io/appium/settings/RecorderService.java

```
@@ -0,0 +1,135 @@
+    private static final String BASE = "io.appium.settings.recording";
+    public static final String ACTION_RECORDING_START = BASE + ".ACTION_START";
+    public static final String ACTION_RECORDING_STOP = BASE + ".ACTION_STOP";
+    public static final String ACTION_RECORDING_RESULT_CODE = "result_code";
+    public static final String ACTION_RECORDING_FILENAME = "recording_filename";
+
+    private static RecorderThread recorderThread;
+
+    public RecorderService() {
+        super();
+    }
+
+    @Override
+    public void onDestroy() {
+        Log.v(TAG, "onDestroy:");
+        if (recorderThread != null && recorderThread.isRecordingContinue()) {
+            recorderThread.stopRecording();
+        }
+        super.onDestroy();
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(final Intent intent) {
+        return null;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    @Override
+    public int onStartCommand(final Intent intent, final int flags, final int startId) {
+        Log.v(TAG, "onStartCommand:intent=" + intent);
+
+        int result = START_STICKY;
+        final String action = intent != null ? intent.getAction() : null;
+        Log.v(TAG, "onStartCommand:action string: " + action);
+        if (ACTION_RECORDING_START.equals(action)) {
+            showNotification(); // TODO is this really necessary
+
+            MediaProjectionManager mMediaProjectionManager =
+                    (MediaProjectionManager) getSystemService(Context.MEDIA_PROJECTION_SERVICE);
+
+            if (mMediaProjectionManager != null) {
+                startScreenRecord(mMediaProjectionManager, intent);
+            } else {
+                Log.e(TAG, "mMediaProjectionManager null");
+            }
+        } else if (ACTION_RECORDING_STOP.equals(action)) {
+            Log.v(TAG, "onStartCommand:intent=" + "stop");
+            stopScreenRecord();
+            result = START_NOT_STICKY;
+        }
```

should we return an error is the action name is unknown?

---

done